components/build/src/lib.rs

```diff
@@ -173,3 +174,13 @@ where
 pub fn get_exonum_protobuf_files_path() -> String {
     env::var("DEP_EXONUM_PROTOBUF_PROTOS").expect("Failed to get exonum protobuf path")
 }
+
+// The problem is that
+// 1) On Windows path can contain both / and \.
+// 2) `protoc_rust` compares paths as strings but same path can have different representation as a string.
+fn clean_path<P: AsRef<Path>>(path: P) -> PathBuf {
```

Comment:
I think that this may be used as a <i>temporary</i>™ workaround, but it sounds like a bug that has to be reported to **rust-protobuf**. I would strongly consider taking time to pinpoint and report it (there doesn't seem to be a similar issue yet). The whole reason behind `std::fs::Path` existence is to prevent exactly this kind of bugs.

I'd also rename this function to `canonicalize_separators` because 'clean' is not obvious enough.

It would also be great to see some basic tests for this function so that its effect is more clear.

I'm also slightly curious whether this implementation correctly handles all intricacies of Windows paths, like disk labels (`C:\Some\Directory` not turning into `/C:/Some/Directory` with a leading slash) and UNC paths (`\\some\path` not turning into `/some/path`, with a single slash). This `to_str` also looks fishy because strictly speaking paths are not required to be valid Unicode text (on Windows and UNIX alike). Another thing that I would like to note is that backslash `\` can be used in file names on UNIX platforms so we probably should not convert it into a directory separator.

However, I'm willing to sacrifice a bit of my karma and can ignore all these minutiae considering that this is a workaround which hopefully will be removed in the future. But I would strongly prefer this to be accompanied by at least a bug report to rust-protobuf (or a fix in rust-protobuf, but that may be to costly to do at the moment), as well as a giant comment `// THIS IS A HACK DON'T DO LIKE THIS IN PRODUCTION CODE KIDS`.